lib/vecmat_external.h

```diff
@@ -87,42 +97,50 @@ constexpr static inline const vector id(ssize_t i = -1)
+constexpr inline vector operator-(const vector other) const { return { x - other.x, y - other.y, z - other.z }; }
+constexpr inline vector operator*(const vector other) const { return { x * other.x, y * other.y, z * other.z }; }
+constexpr inline vector operator/(const vector other) const { return { x / other.x, y / other.y, z / other.z }; }
+
+constexpr inline vector& operator+=(const vector other) { return ((*this) = (*this) + other); }
+constexpr inline vector& operator-=(const vector other) { return ((*this) = (*this) - other); }
+constexpr inline vector& operator*=(const vector other) { return ((*this) = (*this) * other); }
+constexpr inline vector& operator/=(const vector other) { return ((*this) = (*this) / other); }
+
+constexpr inline vector operator+(const scalar s) const { return (*this) + vector{ s, s, s }; }
+constexpr inline vector operator-(const scalar s) const { return (*this) - vector{ s, s, s }; }
+constexpr inline vector operator*(const scalar s) const { return (*this) * vector{ s, s, s }; }
+constexpr inline vector operator/(const scalar s) const { return (*this) / vector{ s, s, s }; }
+
+constexpr inline vector& operator+=(const scalar s) { return ((*this) = (*this) + s); }
+constexpr inline vector& operator-=(const scalar s) { return ((*this) = (*this) - s); }
+constexpr inline vector& operator*=(const scalar s) { return ((*this) = (*this) * s); }
+constexpr inline vector& operator/=(const scalar s) { return ((*this) = (*this) / s); }
+
+friend inline vector operator+(const scalar s, vector rhs) { return  rhs + s; }
+friend inline vector operator-(const scalar s, vector rhs) { return -rhs + s; }
+friend inline vector operator*(const scalar s, vector rhs) { return  rhs * s; }
+friend inline vector operator/(const scalar s, vector rhs) { return (vector{s,s,s}/rhs); }
+
+constexpr inline bool operator==(const vector& other) { return x == other.x && y == other.y && z == other.z; }
+constexpr inline bool operator!=(const vector& other) { return !((*this) == other); }
+constexpr inline bool operator<(const vector& other) { return x < other.x || y < other.y || z < other.z; }
 };
 
-using vector_array = vector;
-
-struct alignas(sizeof(scalar) * 4) aligned_vector {
-using T = scalar;
-constexpr static const size_t N = 3;
-constexpr static const size_t X = 0;
-constexpr static const size_t Y = 1;
-constexpr static const size_t Z = 2;
-union {
-  T xyz[N];
-  struct { T x, y, z; };
-  struct { T r, g, b; };
-};
-constexpr static inline const aligned_vector id(ssize_t i = -1)
-{
-  return aligned_vector{
-  ((i % N) == X) ? (T)1 : (T)0,
-  ((i % N) == Y) ? (T)1 : (T)0,
-  ((i % N) == Z) ? (T)1 : (T)0,
-  };
-}
-constexpr static inline const aligned_vector ne()
-{
-  return aligned_vector{ (scalar)0, (scalar)0, (scalar)0 };
-}
-};
+// zero's out a vector
+static inline void vm_MakeZero(vector *v) { *v = vector::ne(); }
 
+using vector_array = vector;
```

Comment:
why do we need 2 names for the same thing?

Reply:
vector_array is now combined with vector containing xyz and x,y,z and a typedef/use of vector to stay compatible with existing code that uses it.